File: libs/event-schema/src/main/java/gov/cdc/nbs/message/enums/Indicator.java
package gov.cdc.nbs.message.enums;

public enum Indicator {
  YES("Y", "Yes"),
  NO("N", "No"),
  UNKNOWN("UNK", "Unknown");
  private final String code;
  private final String display;

  Indicator(String code, final String display) {
    this.code = code;
    this.display = display;
  }

  public String getId() {
    return code;
  }

  public String getDescription() {
    return display;
  }
}

File: apps/modernization-ui/src/coded/Indicator.ts
interface SelectableIndicator {
    value: string;
    name: string;
}

enum Indicator {
    Yes = 'Y',
    No = 'N',
    Unknown = 'UNK'
}

const indicators = [
    { value: Indicator.Yes, name: 'Yes' },
    { value: Indicator.No, name: 'No' },
    { value: Indicator.Unknown, name: 'Unknown' }
];

export { Indicator, indicators };
export type { SelectableIndicator };
